src/L1Deployer.sol

```diff
@@ -195,6 +201,35 @@ contract L1Deployer is RoleManager {
         emit UpdateEscrowManager(_rollupID, _escrowManager);
     }
 
+    /**
+     * @notice Must be called by the L2's Admin in order to deploy the L2 Deployer contract.
+     */
+    function deployL2Deployer(
+        uint32 _rollupID,
+        address _l2Admin,
+        address _l2RiskManager,
+        address _l2EscrowManager
+    ) external {
+        // Register rollup if not already done. Verifies its a valid rollup ID.
+        if (getRollupContract(_rollupID) == address(0)) {
+            _registerRollup(_rollupID, address(0));
+        }
+
+        // Can only be called by Admin
+        _isRollupAdmin(_rollupID);
+
+        require(_l2Admin != address(0), "ZERO ADDRESS");
+
+        address _l2Factory = getPositionHolder(L2_FACTORY);
+
+        polygonZkEVMBridge.bridgeMessage(
+            _rollupID,
+            _l2Factory,
+            true,
+            abi.encode(_l2Admin, _l2RiskManager, _l2EscrowManager)
```

Comment:
Not required, but maybe make this a struct to make the fields clear on both sides.

Reply:
hmm could do that for the new escrow message too

Do we trust the encoder to properly handle structs?

Reply:
added to the escrow deployment tx https://github.com/yearn/yearn-stb/commit/3e3fea05bd609fbff7a528e33bf44c1c33803796